src/main/java/seedu/duke/module/Module.java

```diff
@@ -0,0 +1,44 @@
+package seedu.duke.module;
+
+import seedu.duke.data.ModuleList;
+
+public class Module {
+    protected String name;
+    protected String id;
+    protected String semester;
+    protected String description;
+    protected Boolean isSUable;
+    protected ModuleList preReqModules;
+    protected boolean isNameValid;
+    protected boolean isIdValid;
+
+
+    /**
+     * This is Module's constructor.
+     * @param type the type of module identifier.
+     * @param identifier user input identifier.
+     */
+    public Module(String type, String identifier, String semester) {
```

Comment:
Maybe the variable name identifier could be more specific?

Reply:
Yeah, but I cannot figure out a good one.😂 The `identifier` indicates users' input and it can be either module's name or module's id(code). Maybe I can change it into `userInputModule` and explain the meaning in the javadoc.

Reply:
Oh sorry I made a mistake. Module can be not only added by users but also added from the database. Maybe here `moduleIdentifier` is better and I will still explain the meaning in the javadoc.